cmd/diff/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/pkg/diff"
+)
+
+var (
+	color       = flag.Bool("color", false, "color")
+	contextSize = flag.Int("context", 3, "context size")
+)
+
+func fileLines(filename string) ([]string, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var lines []string
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		lines = append(lines, s.Text())
+	}
+	return lines, s.Err()
+}
+
+func run() error {
+	ctx := context.Background()
+
+	flag.Parse()
+	if len(flag.Args()) != 2 {
+		return fmt.Errorf("syntax: %s filename1 filename2", os.Args[0])
+	}
+
+	aName := flag.Arg(0)
+	aLines, err := fileLines(aName)
+	if err != nil {
+		return err
+	}
+
+	bName := flag.Arg(1)
+	bLines, err := fileLines(bName)
+	if err != nil {
+		return err
+	}
+
+	ab := diff.Strings(aLines, bLines)
+	e := diff.Myers(ctx, ab).WithContextSize(*contextSize)
+	opts := []diff.WriteOpt{
+		diff.Names(aName, bName),
+	}
+	if *color {
+		opts = append(opts, diff.TerminalColor())
+	}
+	_, err = e.WriteUnified(os.Stdout, ab, opts...)
+	return err
+}
+
+func main() {
+	if err := run(); err != nil {
```

Comment:
I don't think this wrapper buys us much. I'd suggest instead something like:

```func check(err error) {
  if err != nil {
    log.Fatal(err)
  }
}
```

and then turning `func run` into `func main`, and calling `check(err)` as needed.

Reply:
Done.